### PR TITLE
Assignment 4 w140057

### PR DESCRIPTION
**Pull request by RicTer3637**
Added all the changes requested from @anaigmo, created a new pull request due to starting a new branch and the folder changes

### CODE REVIEW

File: Assignment4/RicTer3637-w140057/task06.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+# Visualize the results
+
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JohnSmith, VCARD.EmployedAt, ns.UPM))
```
Comment (dgarijo) [inline Assignment4/RicTer3637-w140057/task06.py]: vcard does not define this property. Either create your own or search for this property in another vocab

File: Assignment4/RicTer3637-w140057/task07.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+# Visualize the results
+
+#for r in g.query(q1):
+#  print(r)
+ns=Namespace("http://somewhere#")
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+ print(s)
```
Comment (dgarijo) [inline Assignment4/RicTer3637-w140057/task07.py:33]: This won't print the subclasses of subclasses of Living thing. Hint: Make the function recursive.

File: Assignment4/RicTer3637-w140057/task07.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+# TO DO
+# Visualize the results
+
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery('''
+  SELECT ?subclass WHERE{
+    ?subclass rdfs:subClassOf/rdfs:subClassOf* ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/RicTer3637-w140057/task07.py]: the operator `*` means 0 or more subclassOf. So what you are writing here is `+` (1 or more). This is not incorrect, but a weird way of stating it.

File: Assignment4/RicTer3637-w140057/task07.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+for sujeto, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+     for sujeto1, p, o in g.triples((None, RDF.type, sujeto)):
+         print(sujeto1)
+
+
+q1 = prepareQuery('''
+SELECT ?sujeto WHERE{
+  {
+    ?sujeto rdf:type ?subclass.
+    optional{?subclass rdfs:subClassOf ?subclass.}
```
Comment (dgarijo) [inline Assignment4/RicTer3637-w140057/task07.py]: You need to use the `*` operator here.

File: Assignment4/RicTer3637-w140057/task07.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+#    results.append((s,p,o))
+#  return results
+
+#for sujeto in lista(ns.Person):
+#  print(sujeto[0])
+
+
+for sujeto, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+     for sujeto1, p, o in g.triples((None, RDF.type, sujeto)):
+         print(sujeto1)
```
Comment (dgarijo) [inline Assignment4/RicTer3637-w140057/task07.py:78]: This is not taking into account the individuals of the subclasses of Person

File: Assignment4/RicTer3637-w140057/task07.py (hunk 4 above)
Comment (dgarijo) [reply Assignment4/RicTer3637-w140057/task07.py]: also this optional is not needed if you use `*`

File: Assignment4/RicTer3637-w140057/task07.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+  if len(set1)>=2 :
+    for elem in set1:
+      print(person)
+for person,p,o in g.triples((None,FOAF.knows,person)):
+  set1=set()
+  for elem in g.triples((person,FOAF.knows,None)):
+    set1.add(elem[2])
+  if len(set1)>=2 :
+    for elem in set1:
+      print(person)
```
Comment (dgarijo) [inline Assignment4/RicTer3637-w140057/task07.py:181]: Missing 7.4 and 7.5 in SPARQL

File: Assignment4/RicTer3637-w140057/task07.py [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+from rdflib import FOAF
+
+for person,p,o in g.triples((person,FOAF.knows,None)):
+  set1=set()
+  for elem in g.triples((person,FOAF.knows,None)):
+    set1.add(elem[2])
+  if len(set1)>=2 :
+    for elem in set1:
+      print(person)
+for person,p,o in g.triples((None,FOAF.knows,person)):
```
Comment (dgarijo) [inline Assignment4/RicTer3637-w140057/task07.py:175]: why do you repeat it twice?

**Review by dgarijo — CHANGES_REQUESTED** [review]
Missing exercises

**Comment by dgarijo** [comment]
Still missing changes on task 7

**Comment by RicTer3637** [comment]
Added changes on task 7

**Comment by anaigmo** [comment]
Entregado con algunos errores sin corregir, no se aceptan más PR

**Pull request merged**